### PR TITLE
feat: add request validation to cache-service

### PR DESCRIPTION
**Pull request by koonopek**

### CODE REVIEW

File: packages/cache-service/src/data-packages/data-packages.controller.ts [hunk 1]
```diff
@@ -11,56 +11,34 @@ import {
...
-    dataServiceId: string;
-  };
-}
-
 const CONTENT_TYPE_OCTET_STREAM = "application/octet-stream";
 const CONTENT_TYPE_TEXT = "text/html";
 const CONTENT_TYPE_JSON = "application/json";
 
 @Controller("data-packages")
+@UsePipes(new ValidationPipe({ transform: true }))
```
Comment (cehali) [inline packages/cache-service/src/data-packages/data-packages.controller.ts:41]: Maybe add it at the application level: https://docs.nestjs.com/techniques/validation#auto-validation ?

Comment (koonopek) [reply packages/cache-service/src/data-packages/data-packages.controller.ts:41]: Then you also have to remember to add it in every test.
Like this you have to only add it one per every controller:
```js
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = moduleFixture.createNestApplication();
    // app.useGlobalPipe(new ValidationPipe())
    await app.init();
```
Also allow to pass custom option per controller thus more granular control.

File: packages/cache-service/src/data-packages/data-packages.interface.ts [hunk 2]
```diff
@@ -1,5 +1,90 @@
...
+  SignedDataPackagePlainObj,
+} from "redstone-protocol";
+import { IsNumberOrString } from "../utils/IsNumberOrString";
+import { CachedDataPackage } from "./data-packages.model";
 
-export interface ReceivedDataPackage extends SignedDataPackagePlainObj {
-  sources?: object;
+export class ReceivedDataPackage implements SignedDataPackagePlainObj {
+  @IsString()
+  signature: string;
```
[the remark this answers is not in the document]
Comment (koonopek) [reply packages/cache-service/src/data-packages/data-packages.interface.ts:19]: We can but then mock-signatures and stuff like this won't work. Do we want to be so strict?

File: packages/cache-service/src/data-packages/data-packages.interface.ts [hunk 3]
```diff
@@ -1,5 +1,90 @@
...
+  @IsOptional()
+  sources?: Record<string, string | number>;
+}
+
+export class ReceivedDataPoint implements IStandardDataPoint {
+  @IsString()
+  dataFeedId: string;
+
+  @Validate(IsNumberOrString)
+  value: string;
```
Comment (cehali) [inline packages/cache-service/src/data-packages/data-packages.interface.ts:37]: If we validate it to be string or number, we should stick with this typing probably?

File: packages/cache-service/src/data-packages/data-packages.interface.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (koonopek) [reply packages/cache-service/src/data-packages/data-packages.interface.ts:19]: done, I hope we wont break something :)

**Review by hatskier — APPROVED** [review]

**Review by hatskier — APPROVED** [review]

**Pull request merged**